grr/server/grr_response_server/gui/ui/components/code_editor/code_editor.ts

```diff
@@ -74,14 +60,155 @@ export class CodeEditor implements AfterViewInit, ControlValueAccessor {
     this.editor = CodeMirror.fromTextArea(this.editorTarget.nativeElement, {
       value: '',
       mode: 'text/x-sqlite',
-      theme: 'idea',
+      theme: 'neo',
       extraKeys: {'Ctrl-Space': 'autocomplete'},
       lineNumbers: true,
       lineWrapping: true,
     });
 
     this.editor.on('change', () => {
       this.announceValueChanged(this.editorValue);
+      this.editorValueChanges$.next(this.editorValue);
     });
   }
 }
+
+/**
+ * Extends the core editor with ControlValueAccessor functoinality.
+ * This allows the code editor to be used as a Angular form field.
+ */
+class EditorWithCva extends CodeEditorCore implements ControlValueAccessor {
+  writeValue(value: string): void {
+    this.editorValue = value;
+  }
+  registerOnChange(fn: OnChangeFn): void {
+    this.announceValueChanged = fn;
+  }
+  registerOnTouched(): void { }
+}
+
+/**
+ * Extends the editor with MatFormFieldControl functionality.
+ * This allows the editor to be used inside <mat-form-field></mat-form-field>
+ * tags.
+ */
+class EditorWithCvaAndMffc extends EditorWithCva
```

Comment:
It's a bit hard to read to have 4 classes, 3 of them are only used once as base class for the next higher up class.  E.g. CodeEditor is only a minimal layer on EditorWithCvaAndMffc and EditorWithCva is only a minimal layer on CoreEditor. I suggest unifying at least those two to simplify reading this code and simplify the naming of the classes.